File: SoftUni Lessons/Python Development/Python Basics April 2019/Lessons and Problems/09.1 - While Loop More Exercises/02. Report System.py
needed_money = int(input())
collected_money = 0
command = None

counter = 0
collected_from_cash = 0
collected_from_card = 0
times_with_cash = 0
times_with_card = 0

while True:
    if collected_money >= needed_money:
        break
    command = input()
    if command == "End":
        break
    counter += 1
    if counter % 2 != 0:
        if int(command) > 100:
            print(f"Error in transaction!")
            continue
        else:
            print(f"Product sold!")
            collected_money += int(command)
            collected_from_cash += int(command)
            times_with_cash +=1
    else:
        if int(command) < 10:
            print(f"Error in transaction!")
            continue
        else:
            print(f"Product sold!")
            collected_money += int(command)
            collected_from_card += int(command)
            times_with_card += 1

if command == "End" and collected_money < needed_money:
    print(f"Failed to collect required money for charity.")
else:
    print(f"Average CS: {(collected_from_cash / times_with_cash):.2f}")
    print(f"Average CC: {(collected_from_card / times_with_card):.2f}")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/14. УПРАЖНЕНИЕ ЦИКЛИ - ЧАСТ 2/While-Loop - More Exercises/02. Report System.js
function reportSystem(params) {
    let aim = Number(params.shift());

    let timesCash = 0;
    let totalCash = 0;

    let timesCard = 0;
    let totalCard = 0;

    let times = 1;
    while ("We colect the money or they say End") {
        let product = params.shift();
        
        if (product == 'End') {
            console.log(`Failed to collect required money for charity.`);
            break;
        }
        
        product = Number(product);
        
        if (times % 2 != 0) {
            // We get paid in cash.
            if (product > 100) {
                console.log(`Error in transaction!`);
            } else {
                aim -= product;
                console.log(`Product sold!`)
                timesCash += 1;
                totalCash += product
            }

        } else {
            // We get paid with card.
            if (product < 10) {
                console.log(`Error in transaction!`);
            } else {
                aim -= product;
                console.log(`Product sold!`)
                timesCard += 1
                totalCard += product
            }
        }

        times += 1;

        if (aim <= 0) {
            console.log(`Average CS: ${(totalCash / timesCash).toFixed(2)}`);
            console.log(`Average CC: ${(totalCard/ timesCard).toFixed(2)}`);
            break;
        }
    }
}

reportSystem([500, 120, 8, 63, 256, 78, 317]);  // Expected Output:
// Error in transaction!
// Error in transaction!
// Product sold!
// Product sold!
// Product sold!
// Product sold!
// Average CS: 70.50
// Average CC: 286.50

reportSystem([600, 86, 150, 98, 227, 'End']);  // Expected Output:
// Product sold!
// Product sold!
// Product sold!
// Product sold!
// Failed to collect required money for charity.
